src/adapter/graphql/src/mutations/dataset_mut.rs

```diff
@@ -91,7 +91,7 @@ impl DatasetMut {
     async fn delete(&self, ctx: &Context<'_>) -> Result<DeleteResult> {
         let delete_dataset = from_catalog::<dyn domain::DeleteDatasetUseCase>(ctx).unwrap();
         match delete_dataset
-            .execute_via_handle(&self.dataset_handle)
+            .execute_via_handle(&self.dataset_handle, false)
```

Comment:
We might need to accept `force` parameter via API argument. I.e., UI will ask to confirm.